typing/typeclass.ml

```diff
@@ -1631,7 +1634,17 @@ let class_description env sexpr =
   (expr, expr.cltyp_type)
 
 let class_declarations env cls =
-  type_classes true approx_declaration class_declaration env cls
+  let info, env =
+    type_classes true approx_declaration class_declaration env cls
+  in
+  let ids, exprs =
+    List.split
+      (List.map
+         (fun (id, _, _, _, _, _, _, _, _, _, _, {ci_expr}) -> id, ci_expr)
```

Comment:
We should not have to write this. What about some recordification work in `typeclass.ml`?